openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -1217,3 +1217,18 @@ def component_version_asset(request, component_version_uuid, asset_path):
         content.read_file().chunks(),
         headers=redirect_response.headers,
     )
+
+
+@require_safe
+def component_draft_asset(request, usage_key_str, asset_path):
+    """
+    Serves the draft version of static assets associated with a Library Component.
+
+    See `component_version_asset` for more details
+    """
+    key = LibraryUsageLocatorV2.from_string(usage_key_str)
```

Comment:
Please use the more general `UsageKeyV2` instead of the more specific `LibraryUsageLocatorV2`, and catch `InvalidKeyError` (return a 400 for it).

Reply:
Or you can use the new `UsageKeyV2Converter` from https://github.com/openedx/edx-platform/pull/35676/files#diff-2db308e15c5d2a16d52e1c0ef32ec7292fdff4f96a025f4e5c0a390ba2a585c3 which will be merging shortly.